packages/xstate-vue/src/useActorRef.ts

```diff
@@ -0,0 +1,43 @@
+import { onBeforeUnmount, onMounted } from 'vue';
+import {
+  ActorRefFrom,
+  AnyActorLogic,
+  AnyStateMachine,
+  AreAllImplementationsAssumedToBeProvided,
+  InternalMachineImplementations,
+  createActor,
+  ActorOptions,
+  Observer,
+  SnapshotFrom,
+  StateFrom,
+  Subscription,
+  toObserver
+} from 'xstate';
+
+export type UseActorRefRestParams<TLogic extends AnyActorLogic> = [
+  options?: ActorOptions<TLogic>,
+  observerOrListener?:
+    | Observer<SnapshotFrom<TLogic>>
+    | ((value: SnapshotFrom<TLogic>) => void)
+];
+
+export function useActorRef<TLogic extends AnyActorLogic>(
+  actorLogic: TLogic,
+  ...[options = {}, observerOrListener]: UseActorRefRestParams<TLogic>
+): ActorRefFrom<TLogic> {
+  const actorRef = createActor(actorLogic, options).start();
```

Comment:
This should be moved to `onMounted`. React is not unique with its SSR constraints - we should follow the same rules in each framework.